tests/contracts/integratees/utils/MockIntegrateeBase.sol

```diff
@@ -69,11 +70,48 @@ abstract contract MockIntegrateeBase is NormalizedRateProviderBase {
             require(asset != address(0), "__swap: empty value in _assetsFromIntegratee");
             require(amount > 0, "__swap: empty value in _assetsFromIntegrateeAmounts");
             if (asset == ETH_ADDRESS) {
-                msg.sender.transfer(amount);
+                _trader.transfer(amount);
             }
             else {
-                ERC20(asset).transfer(msg.sender, amount);
+                ERC20(asset).transfer(_trader, amount);
             }   
         }
     }
+
+    function __getRateAndSwapAssets(
```

Comment:
I see why you would want to lift this logic, but I don't think this should be in `MockIntegrateeBase`, because some integratees will have multiple incoming or outgoing assets (0x, Uniswap LPs, etc).

It's true that several can use this exact same function though. Maybe create a `MockSimpleIntegrateeBase`?